src/components/searchbar/searchbar.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState} from 'react';
+import { graphql, useStaticQuery } from "gatsby";
+import './searchbar.css'
+import { SearchIcon } from "../footer";
+import {  RiCloseFill } from "react-icons/ri";
+
+
+const SearchBar = (placeholder: any): JSX.Element => {
+    const data = useStaticQuery(graphql`
+    query {
+        allMdx {
+          nodes {
+            frontmatter {
+              slug
+              title
+            }
+          }
+        }
+      }
+    `);
+    const [filteredData, setFilteredData] = useState([]);
+    const [wordEntered, setWordEntered] = useState("");
```

Comment:
Type cast according to the state types
```suggestion
    const [filteredData, setFilteredData] = useState<typeof data[]>();
    const [wordEntered, setWordEntered] = useState<string>("");
```

Reply:
done